src/Servers/Kestrel/Core/src/Internal/Infrastructure/StringUtilities.cs

```diff
@@ -109,6 +112,127 @@ public static unsafe bool TryGetAsciiString(byte* input, char* output, int count
             return isValid;
         }
 
+        [MethodImpl(MethodImplOptions.AggressiveOptimization)]
+        public unsafe static bool BytesOrdinalEqualsStringAndAscii(string previousValue, Span<byte> newValue)
+        {
+            // We just widen the bytes to char for comparision, if either the string or the bytes are not ascii
+            // this will result in non-equality, so we don't need to specifically test for non-ascii.
+            Debug.Assert(previousValue.Length == newValue.Length);
+
+            // Use IntPtr values rather than int, to avoid unnessary 32 -> 64 movs on 64-bit.
```

Comment:
nit: "unnessary" => "unnecessary "
```suggestion
            // Use IntPtr values rather than int, to avoid unnecessary 32 -> 64 movs on 64-bit.
```

Reply:
Done